build-tools/packages/build-cli/src/handlers/checkFunctions.ts

```diff
@@ -477,6 +478,91 @@ export const checkAssertTagging: StateHandlerFunction = async (
 	return true;
 };
 
+/**
+ * Checks that release notes have been generated.
+ *
+ * @param state - The current state machine state.
+ * @param machine - The state machine.
+ * @param testMode - Set to true to run function in test mode.
+ * @param log - A logger that the function can use for logging.
+ * @param data - An object with handler-specific contextual data.
+ * @returns True if the state was handled; false otherwise.
```

Comment:
These docs seems like they belong on the definition of StateHandlerFunction, and not on every instance of it.

Also these docs don't provide enough contexts to someone who doesn't already know this code well to evaluate correctness. From the description "Set to true to run function in test mode" I don't know how to evaluate if the implementation here correctly handles test mode, since what test mode is is unclear.

If there are a bunch of things in this codebase which refer to test mode and should share documentation, consider making testMode an enum instead of a Boolean so you can provide detailed documentation on its declaration which will be easy to find for anyone looking at this code. If its not used elsewhere, provide more detail here (well on StateHandlerFunction)

Reply:
Yeah, the docs are non-existent. My hope is that reviewers can see the surrounding code and judge whether this function
is like the others, since they're all the same shape. I added some more context to the StateHandlerFunction definition.

Not that it's helpful here, but "test mode" is used to test that all states of the state machine are handled. It
protects against adding a new state without code that at least pretends to handle it.